### DISCUSSION TITLE
seaborn.objects: Rotation of ticks

### DISCUSSION THREAD

**Discussion by JanPalasek** [mwaskom/seaborn#3169] [Q&A]
Hello,

might I suggest a new feature request: rotation of the ticks for seaborn objects? Or is it possible at this moment? I haven't found it yet.

Currently I use `.on(axis).plot()` and then on this axis I set it up via matplotlib with `axis.xaxis.set_tick_params(rotation=90)`.

**Comment by mwaskom** [comment]
I assume you mean rotation of tick _labels_? I don't think that's currently supported since it's not (AFAICT) something you can set a default for with `rcParams`.

**Comment by JanPalasek** [comment]
> I assume you mean rotation of tick _labels_? I don't think that's currently supported since it's not (AFAICT) something you can set a default for with `rcParams`.

I presume the implementation is difficult then. That's pity, data scientists from my bubble need it quite often. If there was a switch for example `so.Continuous().label(rotation=90)`, it would be awesome.

**Comment by shoaibkhanz** [comment]
@JanPalasek quite naturally, I tried something very similar, it could also be part of the tick method    `fig.scale(x=so.Nominal().tick())`

**Comment by mwaskom** [comment]
That's a reasonable place expect the parameter, but implementation-wise, it doesn't make sense. `Scale.tick` and `Scale.format` are interfaces to define matplotlib `Locator`/`Formatter` objects. But those can't control the label rotation, which is a property of the label _artist_.

**Comment by JanPalasek** [comment]
@mwaskom Could this interface be expanded?

```python
def label(self, rotation=None,  formatter_param1=None, formatter_param2=None, formatter=None):
  pass
```

Or even decide on each individual parameter, whether it is a parameter for formatter or for some other component (artist?).

**Comment by Demetrio92** [comment]
`seaborn.axisgrid.FacetGrid` has a method `.set_xticklabels` with an option to rotate the labels. However, on my example it's currently buggy #3341

**Comment by GStechschulte** [comment]
See the answer by Brian Head in this StackOverflow [response](https://stackoverflow.com/questions/74715767/how-to-rotate-the-xticks-with-seaborn-objects). 

It has worked well in plots with faceting and or no-facets.

```python3
p = (
    so.Plot(data=df, x="...", y="...")
    .facet(col="...")
    .add(so.Bars())
    .theme({"figure.figsize": (16, 6)})
    ).plot() # Notice the call to plot

# If the facet includes two columns
p._figure.axes[0].xaxis.set_tick_params(rotation=90)
p._figure.axes[1].xaxis.set_tick_params(rotation=90)
p.show()
```

**Reply by timslittle** [reply]
Great, thanks! For people with multiple columnar facets this should iterate over each:
```python
for axis in p._figure.axes:
    axis.xaxis.set_tick_params(rotation=90)
```
